src/file_utils.cr

```diff
@@ -384,7 +384,17 @@ module FileUtils
       end
       Dir.delete(path)
     else
-      File.delete(path)
+      {% if flag?(:win32) %}
+        begin
+          File.delete(path)
+        rescue File::AccessDeniedError
+          # To be able to delete read-only files (e.g. ones under .git/) on Windows.
+          File.chmod(path, 0o666)
+          File.delete(path)
```

Comment:
Maybe if this delete fails again, restore previous chmod and then raise?

Reply:
Well, it *does raise*, but I think restoring the original permissions would be super difficult.

Reply:
As far as the current standard library is concerned, the only change needed is `LibC.SetFileAttributesW` because [we probably won't be touching the other permissions in Crystal](https://github.com/crystal-lang/crystal/issues/13233). Also it can be done in `Crystal::System::File.delete` directly rather than only in `rm_r`